File: apps/backend/handlers/nfts.go
package handlers

import (
	"encoding/json"
	"fmt"
	"net/http"
	"neura-nft/services"
)

func ExploreNFTsHandler(w http.ResponseWriter, r *http.Request) {
	address := r.URL.Query().Get("address")
	category := r.URL.Query().Get("category")

	if address == "" || category == "" {
		http.Error(w, "missing address or category", http.StatusBadRequest)
		return
	}

	owned, trending, err := services.GetExploreNFTs(address, category)
	if err != nil {
		fmt.Println("❌ Error in GetExploreNFTs:", err)
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	if err := json.NewEncoder(w).Encode(map[string]any{
		"owned":    owned,
		"trending": trending,
	}); err != nil {
		fmt.Println("❌ JSON encode error:", err)
	}
}

File: apps/backend/src/controllers/nfts.ts
import { getExploreNFTs } from "@/services/nfts";
import { Request, Response } from "express";

export const exploreNftsHandler = async (req: Request, res: Response) => {
  const { address, category } = req.query;

  if (
    !address ||
    !category ||
    typeof address !== "string" ||
    typeof category !== "string"
  ) {
    res.status(400).json({ message: "missing address or category" });
    return;
  }

  const { owned, trending } = await getExploreNFTs(address, category);
  res.json({ owned, trending });
};
